cli.js

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env node
+var fs = require('fs')
+var minimist = require('minimist')
+var fastq = require('./index')
+
+var argv = minimist(process.argv.slice(2), {
+  alias: {
+    help: 'h'
+  }
+})
+
+if (argv.help || argv._.length === 0) {
+  console.log(
+        'Usage: bionode-fastq <options> <fastq file [required]> <output file>\n' +
+        'If no output is provided, the result will be printed to stdout\n'
+    )
+  process.exit(0)
+}
+
+// if there is an output file present, write to the file. Otherwise write to stdout
+var output = argv._[1] ? fs.createWriteStream(argv._[1]) : process.stdout
+
+var fq = fastq.read(argv._[0])
+
+process.stdin.setEncoding('utf8')
+
+fq.on('data', function (data) {
```

Comment:
@thejmazz I am very new to nodejs streams, so I probably just don't understand the concept behind pipes. But, would using the code that you referenced above require the parser to be refactored? From what I understand, `fastq.read(argv._[0])` returns an `events.EventEmitter()` object, so can an `events.EventEmitter()` be streamed?

Perhaps this is the function of the `ndjson` module and I just don't know it.